src/app/leverage/provider.tsx

```diff
@@ -239,6 +245,51 @@ export function LeverageProvider(props: { children: any }) {
     fetchStats()
   }, [baseToken])
 
+  useEffect(() => {
+    async function fetchCostOfCarry(
+      jsonRpcProvider: providers.JsonRpcProvider,
+    ) {
+      try {
+        const poolDataProviderContract = new UiPoolDataProvider({
+          uiPoolDataProviderAddress: AaveV3Ethereum.UI_POOL_DATA_PROVIDER,
+          provider: jsonRpcProvider,
+          chainId: ChainId.mainnet,
+        })
+        const reserves = await poolDataProviderContract.getReservesHumanized({
+          lendingPoolAddressProvider: AaveV3Ethereum.POOL_ADDRESSES_PROVIDER,
+        })
+
+        const formattedPoolReserves = formatReserves({
+          reserves: reserves.reservesData,
+          currentTimestamp: Math.floor(Date.now() / 1000),
+          marketReferenceCurrencyDecimals:
+            reserves.baseCurrencyData.marketReferenceCurrencyDecimals,
+          marketReferencePriceInUsd:
+            reserves.baseCurrencyData.marketReferenceCurrencyPriceInUsd,
+        })
+
+        const borrowedAsset = formattedPoolReserves.find(
+          (asset) =>
+            asset.symbol.toLowerCase() === outputToken.borrowedAssetSymbol,
+        )
+
+        if (!borrowedAsset) {
+          return
+        }
+
+        setCostOfCarry(
+          Number(borrowedAsset.variableBorrowAPY) -
+            Number(borrowedAsset.supplyAPY),
+        )
+      } catch (e) {
+        console.error('Caught error while fetching borrow rates', e)
+      }
+    }
+
+    if (!jsonRpcProvider || outputToken === null) return
+    fetchCostOfCarry(jsonRpcProvider)
+  }, [jsonRpcProvider, outputToken])
```

Comment:
Please use jsonRpcProvider like [here](https://github.com/IndexCoop/index-app/pull/1238#discussion_r1600468064) as well because this was causing some re-rendering issues. So initializing the provider inside the function from publicClient.

Reply:
@janndriessen good call - done 👍